File: Code/taskman-api-logic/src/main/java/shipreq/taskman/api/TaskType.java
package shipreq.taskman.api;

// TODO Fuck this, just use a sealed trait myself and fail-fast on dup ids

public enum TaskType {

    RegistrationRequested(100),
    RegistrationCompleted(101),
    ReRegistrationAttempted(102),
    PasswordResetRequested(103),
    LandingPageHit(200);

    public final int id;
    private TaskType(int id) {
        this.id = id;
    }
}

File: Code/taskman-api-logic/src/main/scala/shipreq/taskman/api/TaskType.scala
package shipreq.taskman.api

import japgolly.microlibs.adt_macros.AdtMacros
import japgolly.univeq.UnivEq

sealed trait TaskType
object TaskType {
  case object DummyTask               extends TaskType
  case object LandingPageHit          extends TaskType
  case object PasswordResetRequested  extends TaskType
  case object RegistrationCompleted   extends TaskType
  case object RegistrationRequested   extends TaskType
  case object ReportClientError       extends TaskType
  case object ReportServerError       extends TaskType
  case object ReRegistrationAttempted extends TaskType
  case object SendDiagEmail           extends TaskType
  case object SyncToMailingList       extends TaskType
  case object UserFeedbackReceived    extends TaskType
  case object UserUpdated             extends TaskType

  implicit def univEq: UnivEq[TaskType] = UnivEq.derive

  val values = AdtMacros.adtValues[TaskType]
}
